certification/internal/shell/has_license.go

```diff
@@ -1,30 +1,43 @@
 package shell
 
 import (
-	"os/exec"
 	"strings"
 
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
 	log "github.com/sirupsen/logrus"
 )
 
 type HasLicenseCheck struct{}
 
 func (p *HasLicenseCheck) Validate(image string) (bool, error) {
-	stdouterr, err := exec.Command("podman", "run", "-it", "--rm", "--entrypoint", "ls", image, "-A", "/licenses").CombinedOutput()
-	result := string(stdouterr)
+
+	return p.validate(image)
+}
+
+func (p *HasLicenseCheck) validate(image string) (bool, error) {
+	runOpts := cli.ImageRunOptions{
```

Comment:
Great start. But let's put the podman stuff into getDataToValidate like in base_on_ubi.